File: 프로그래머스/1/135808. 과일 장수/과일 장수.js
function solution(k, m, score) {
    
    score.sort((a,b) => a-b);
    
    var min = k;
    
    var answer = 0;
    
    while(score.length >= m){
        for(let i=0; i<m; i++){
            min = Math.min(min, score.pop());
        }
        answer += min*m;
    }
    
    return answer;
}

File: 프로그래머스/1/135808. 과일 장수/과일 장수.java
import java.util.*;

class Solution {
    public int solution(int k, int m, int[] score) {
        
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());
        
        for(int i=0; i<score.length; i++){
            pq.add(score[i]);
        }
        
        int max = k;
        int answer = 0;
        
        while(pq.size() >= m){
            for(int i=0; i<m; i++){
                max = Math.min(max, pq.poll());
            }
            answer += max*m;
        }
        
        return answer;
    }
}